pkg/gadgets/dns/gadget.go

```diff
@@ -193,7 +201,7 @@ func (t *Trace) Start(trace *gadgetv1alpha1.Trace) {
 	attachContainerFunc := func(container *pb.ContainerDefinition) error {
 		key := genKey(container)
 
-		err = t.tracer.Attach(key, container.Pid, newDNSRequestCallback(key))
+		err = t.tracer.Attach(key, container.Pid, newDNSRequestCallback(key), errFunc(key))
```

Comment:
Nit (not really important): execsnoop (and other gadgets) are using a single callback that receives the whole event, so the producer can always call the same callback. 

https://github.com/kinvolk/inspektor-gadget/blob/f3ca079d2ba2522fa63bce2c3b44d32144c07a16/pkg/gadgets/execsnoop/tracer/core/tracer.go#L52-L53

I'm not sure if this is possible to implement here, perhaps it could be useful too.

Reply:
>  so that the tracer packages of each gadgets could be reused in external projects

I was thinking the same and IMO having an event callback that receives a proper defined type make it easier as users will receive the full event, so they know for sure what information the gadget provides.